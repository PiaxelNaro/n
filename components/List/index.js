// @flow
import React from 'react';
import classnames from 'classnames';
import styled from 'styled-components';
import ChevronRightIcon from 'react-icons/lib/md/chevron-right';
import Text from '../Text';
import FlexGrid from '../FlexGrid';
import Button from '../Button';
import Overlay from '../Overlay';
import DropdownButton from '../Button/Dropdown';

type ChoiceList = Array<{ id: string, title: string, subList?: ChoiceList }>;

/**
 * get id form list
 */
function getTitleById(list: ChoiceList, id: string): string {
    const result = list
        .reduce((memo, item) => {
            memo.push({
                id: item.id,
                title: item.title,
            });
            if (item.subList) {
                memo.push(...item.subList);
            }
            return memo;
        }, [])
        .find(item => item.id === id);
    return result ? result.title : '';
}

/**
 * <ListItem/>
 * The basic item component
 * @param {boolean} isActive
 */
const ListItem = styled.a.attrs({
    className: props =>
        classnames({
            'is-active': props.isActive,
            'is-locked': props.isLocked,
        }),
})`
    display: block;
    padding: 5px;
    line-height: 16px;
    border-top: 1px solid rgba(255, 255, 255, 0.6);
    border-left: 1px solid rgba(255, 255, 255, 0.6);
    border-right: 1px solid rgba(0, 0, 0, 0.05);
    border-bottom: 1px solid rgba(0, 0, 0, 0.05);
    /* default */
    cursor: pointer;
    background-color: #F6F6F6;
    &:hover,
    &:focus {
        color: #333;
        background-color: #EFEFEF;
    }
    &.is-active {
        color: #FFF;
        background-color: #1F3149;
        border: 1px solid #1F3149;
        &:hover,
        &:focus {
            color: #FFF;
            background: #1F3149;
        }
    }
    &.is-locked {
        &:hover,
        &:focus {
            color: #BBB;
            background-color: #EFEFEF;
            cursor: default;
        }
    }
`;
ListItem.displayName = 'ListItem';

/**
 * <SubList/>
 * @param {Array}       items
 * @param {Object}      current             active item
 * @param {String}      activeSubListId     check expand/collapse subList
 * @param {Function}    onSelect            click item and callback
 * @param {Function}    onClickSubPanel     toggle subList icon and callback
 */
const SubList = ({
    items,
    current,
    activeSubListId,
    onSelect,
    onClickSubPanel,
}: {
    items: ChoiceList,
    current: string,
    activeSubListId?: string,
    onSelect: string => void,
    onClickSubPanel?: (string, ChoiceList) => void,
}) => {
    const list = items.map(item => {
        let icon;
        if (item.subList) {
            const isCollapsable = item.id !== activeSubListId;
            icon = (
                <Button.Blank
                    className="AASubList-btn"
                    onClick={() => onClickSubPanel(item.id, item.subList)}
                >
                    <SubList.Icon
                        className="AASubList-icon"
                        isCollapsable={isCollapsable}
                    >
                        <ChevronRightIcon size={16} />
                    </SubList.Icon>
                </Button.Blank>
            );
        }
        return (
            <FlexGrid key={item.id}>
                <FlexGrid.Item flex={1}>
                    <ListItem
                        isActive={item.id === current}
                        onClick={() => onSelect(item.id)}
                        className="AASubList-item"
                    >
                        <Text ellipsis>{item.title}</Text>
                    </ListItem>
                </FlexGrid.Item>
                <FlexGrid.Item flex={0}>
                    {icon}
                </FlexGrid.Item>
            </FlexGrid>
        );
    });
    return <div> {list} </div>;
};

SubList.Icon = styled.div`
    width: 40px;
    padding: 5px;
    text-align: left;
    background: ${props => (props.isCollapsable ? 'rgb(246, 246, 246)' : '#AAA')};
    color: ${props => (props.isCollapsable ? '#1F3149' : '#FFF')};
`;

const AAList = styled(FlexGrid)`
    padding: 10px;
    background: url(/static/list-bg.png) repeat-x;
    align-items: flex-start;
`;

/**
 * <ListOverlay />
 * @param {Object}      current     current active list item
 * @param {Array}       list
 * @param {Function}    onSelect    toggle item and callback
 * @param {Number}      itemWidth   item width
 */
class ListOverlay extends React.Component {
    props: {
        current: string,
        list: ChoiceList,
        onSelect: string => void,
        itemWidth: number,
    };

    state: {
        activeSubListId: string,
        subList: ChoiceList,
    };

    onClickSubPanel: Function;
    onSelectItem: Function;
    overlay: { closeOverlay: () => void };

    constructor() {
        super();
        this.state = {
            activeSubListId: '',
            subList: [],
        };
        this.onClickSubPanel = this.onClickSubPanel.bind(this);
        this.onSelectItem = this.onSelectItem.bind(this);
    }

    componentDidMount() {
        this.props.list.map(item => {
            if (item.subList && item.subList.find(sub => sub.id === this.props.current)) {
                this.setState({
                    activeSubListId: item.id,
                    subList: item.subList,
                });
            }
        });
    }

    onSelectItem(id: string) {
        this.props.onSelect(id);
        this.setState({
            activeSubListId: '',
            subList: [],
        });
    }

    onClickSubPanel(id: string, list: ChoiceList) {
        this.setState({
            activeSubListId: id,
            subList: list,
        });
    }

    render() {
        const { current, list, itemWidth } = this.props;

        let subPanel;
        if (this.state.subList.length) {
            subPanel = (
                <div style={{ width: itemWidth }}>
                    <SubList
                        items={this.state.subList}
                        current={current}
                        onSelect={this.onSelectItem}
                    />
                </div>
            );
        }

        return (
            <Overlay>
                <AAList gutter="5px">
                    <div style={{ width: itemWidth }}>
                        <SubList
                            items={list}
                            current={current}
                            onSelect={this.onSelectItem}
                            onClickSubPanel={this.onClickSubPanel}
                            activeSubListId={this.state.activeSubListId}
                        />
                    </div>
                    {subPanel}
                </AAList>
            </Overlay>
        );
    }
}

class List extends React.Component {
    props: {
        current: string,
        label?: string,
        labelIcon?: any,
        list: ChoiceList,
        onSelect: string => void,
        itemWidth: number,
    };

    onSelectItem: Function;
    overlay: { closeOverlay: () => void };

    constructor() {
        super();

        this.onSelectItem = this.onSelectItem.bind(this);
    }

    onSelectItem(id: string) {
        this.overlay.closeOverlay();
        this.props.onSelect(id);
    }

    render() {
        const { current, label, labelIcon, list, itemWidth } = this.props;
        const title = getTitleById(list, current);

        return (
            <Overlay.Trigger
                trigger="click"
                ref={ref => (this.overlay = ref)}
                getOverlay={() =>
                    <ListOverlay
                        current={current}
                        list={list}
                        onSelect={this.onSelectItem}
                        itemWidth={itemWidth}
                    />}
            >
                <DropdownButton label={label} labelIcon={labelIcon} value={title} />
            </Overlay.Trigger>
        );
    }
}

export default List;
export { SubList, ListItem, ListOverlay };
