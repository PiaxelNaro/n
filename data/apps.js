export default [
    {
        id: 1,
        icon:
            'https://static-s.aa-cdn.net/img/gp/20600003867506/WjPDaHXE8ufMDQ43x8qpLfH9yuw1D0mlKzOJLSbeETcRLCONpBChskUtj3DtVTDJknw=w300_w80',
        company: 'Networkcanal Multimedia Ltd',
        name: 'Fa',
    },
    {
        id: 2,
        icon:
            'https://static-s.aa-cdn.net/img/gp/20600004729627/fFI2bIE44LYBwecU17bVeIdvKbRao6O3zek6U6x-5A22uXbEO4fWm15dfffg2jHBTg=w300_w80',
        company: 'KAPS Development',
        name: 'FA',
    },
    {
        id: 3,
        icon:
            'https://static-s.aa-cdn.net/img/ios/1039182706/18de8684db7bcaebb95232d7dc66f252_w80',
        company: 'INNERGY CO.,LTD',
        name: '와이드앵글 FA 교육',
    },
    {
        id: 4,
        icon:
            'https://static-s.aa-cdn.net/img/ios/358534918/ec96f2e8d1fdb23f5cea8671d9d8335a_w80',
        company: '7700 Software',
        name: 'FA World',
    },
    {
        id: 5,
        icon:
            'https://static-s.aa-cdn.net/img/ios/688824991/b3ca3dbc0fb23363af7aa7917d4728a8_w80',
        company: 'Lubomir Havran BINARTS',
        name: 'FA Magazín',
    },
    {
        id: 6,
        icon:
            'https://static-s.aa-cdn.net/img/ios/805530850/d5b350ceb68604faafe6a3c234b07bb7_w80',
        company: 'Tobit',
        name: 'FA/KE',
    },
    {
        id: 7,
        icon:
            'https://static-s.aa-cdn.net/img/ios/884096407/1f6d3ba08a891906570ad2bb68b8dcdd_w80',
        company: 'Tobit',
        name: 'Fa. Handelsagent',
    },
    {
        id: 8,
        icon:
            'https://static-s.aa-cdn.net/img/ios/892841769/b27b0d7ccf10c6182fcf6ddf30017421_w80',
        company: 'Spinn Labs Private Limited',
        name: 'Bao Fa',
    },
    {
        id: 9,
        icon:
            'https://static-s.aa-cdn.net/img/ios/906627580/f41757b0e6c2509e20a937e80e2e0233_w80',
        company: 'E Sports Publications Limited',
        name: 'Lincolnshire FA',
    },
    {
        id: 10,
        icon:
            'https://static-s.aa-cdn.net/img/ios/919412577/ad1ae851ddf94c080bc93ad41c0d20fe_w80',
        company: 'Tobit',
        name: 'FA Foto',
    },
    {
        id: 11,
        icon:
            'https://static-s.aa-cdn.net/img/gp/20600003867506/WjPDaHXE8ufMDQ43x8qpLfH9yuw1D0mlKzOJLSbeETcRLCONpBChskUtj3DtVTDJknw=w300_w80',
        company: 'Networkcanal Multimedia Ltd',
        name: 'Fa1',
    },
    {
        id: 12,
        icon:
            'https://static-s.aa-cdn.net/img/gp/20600004729627/fFI2bIE44LYBwecU17bVeIdvKbRao6O3zek6U6x-5A22uXbEO4fWm15dfffg2jHBTg=w300_w80',
        company: 'KAPS Development',
        name: 'FA1',
    },
    {
        id: 13,
        icon:
            'https://static-s.aa-cdn.net/img/ios/1039182706/18de8684db7bcaebb95232d7dc66f252_w80',
        company: 'INNERGY CO.,LTD',
        name: '와이드앵글 FA 교육1',
    },
    {
        id: 14,
        icon:
            'https://static-s.aa-cdn.net/img/ios/358534918/ec96f2e8d1fdb23f5cea8671d9d8335a_w80',
        company: '7700 Software',
        name: 'FA World1',
    },
];
