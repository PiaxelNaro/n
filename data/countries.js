const countries = {
    ALL: {
        code: 'ALL',
        name: 'All Countries',
    },
    AU: {
        code: 'AU',
        name: 'Australia',
        isLocked: true,
    },
    US: {
        code: 'US',
        name: 'America',
    },
    KO: {
        code: 'KO',
        name: 'South of Korea',
        isLocked: true,
    },
    LG: {
        code: 'LG',
        name: 'Long Long Name Country',
    },
    UK: {
        code: 'UK',
        name: 'England',
    },
};

const categories = [
    {
        name: 'All Countries',
        countries: ['ALL', 'AU', 'US', 'KO', 'LG', 'UK'],
    },
    {
        name: 'Most Popular',
        countries: ['ALL', 'AU', 'US', 'UK'],
    },
];

export { categories, countries };
