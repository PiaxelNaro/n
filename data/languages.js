export default [
    {
        code: 'en-US',
        label: 'English',
    },
    {
        code: 'ja-JP',
        label: '日本語',
    },
    {
        code: 'ko-KR',
        label: '한국어',
    },
    {
        code: 'zh-CN',
        label: '中文(简体)',
    },
    {
        code: 'ru-RU',
        label: 'Русский',
    },
    {
        code: 'fr-FR',
        label: 'Français',
    },
    {
        code: 'de-DE',
        label: 'Deutsch',
    },
];
