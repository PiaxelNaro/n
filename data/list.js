export default {
    current: 'games',
    list: [
        {
            title: 'Overall',
            id: 'overall',
        },
        {
            title: 'Games',
            id: 'games',
            subList: [
                {
                    title: 'Action',
                    id: 'action',
                },
                {
                    title: 'Adventure',
                    id: 'adventure',
                },
                {
                    title: 'Arcade',
                    id: 'arcade',
                },
                {
                    title: 'Board',
                    id: 'board',
                },
                {
                    title: 'Card',
                    id: 'card',
                },
                {
                    title: 'Casino',
                    id: 'casino',
                },
                {
                    title: 'Dice',
                    id: 'dice',
                },
                {
                    title: 'Education',
                    id: 'game_education',
                },
                {
                    title: 'Family',
                    id: 'family',
                },
                {
                    title: 'Kids',
                    id: 'game_kids',
                },
                {
                    title: 'Music',
                    id: 'game_music',
                },
                {
                    title: 'Puzzle',
                    id: 'puzzle',
                },
                {
                    title: 'Racing',
                    id: 'racing',
                },
                {
                    title: 'Role Playing',
                    id: 'role_playing',
                },
                {
                    title: 'Simulation',
                    id: 'simulation',
                },
                {
                    title: 'Sports',
                    id: 'sports',
                },
                {
                    title: 'Strategy',
                    id: 'strategy',
                },
                {
                    title: 'Trivia',
                    id: 'trivia',
                },
                {
                    title: 'Word',
                    id: 'word',
                },
            ],
        },
        {
            title: 'Applications',
            id: 'applications',
        },
        {
            title: 'Books',
            id: 'books',
        },
        {
            title: 'Business',
            id: 'business',
        },
        {
            title: 'Catalogs',
            id: 'catalogs',
        },
        {
            title: 'Education',
            id: 'education',
        },
        {
            title: 'Entertainment',
            id: 'entertainment',
        },
        {
            title: 'Finance',
            id: 'finance',
        },
        {
            title: 'Food and Drink',
            id: 'food_and_drink',
        },
        {
            title: 'Health and Fitness',
            id: 'health_and_fitness',
        },
        {
            title: 'Kids',
            id: 'kids',
            subList: [
                {
                    title: 'Kids 5 & Under',
                    id: 'kids_5_under',
                },
                {
                    title: 'Kids Age 6-8',
                    id: 'kids_age_6_8',
                },
                {
                    title: 'Kids Age 9-11',
                    id: 'kids_age_9_11',
                },
            ],
        },
        {
            title: 'Lifestyle',
            id: 'lifestyle',
        },
        {
            title: 'Magazines and Newspapers',
            id: 'magazines_and_newspapers',
        },
        {
            title: 'Medical',
            id: 'medical',
        },
        {
            title: 'Music',
            id: 'music',
        },
        {
            title: 'Navigation',
            id: 'navigation',
        },
        {
            title: 'News',
            id: 'news',
        },
        {
            title: 'Photo and Video',
            id: 'photo_and_video',
        },
        {
            title: 'Productivity',
            id: 'productivity',
        },
        {
            title: 'Reference',
            id: 'reference',
        },
        {
            title: 'Shopping',
            id: 'shopping',
        },
        {
            title: 'Social Networking',
            id: 'social_networking',
        },
        {
            title: 'Sports',
            id: 'sports',
        },
        {
            title: 'Travel',
            id: 'travel',
        },
        {
            title: 'Utilities',
            id: 'utilities',
        },
        {
            title: 'Weather',
            id: 'weather',
        },
    ],
};
