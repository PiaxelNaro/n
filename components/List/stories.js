// @flow
import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import FolderIcon from 'react-icons/lib/fa/folder-open-o';
import GlobeIcon from 'react-icons/lib/fa/globe';
import List from '.';
import FlexGrid from '../FlexGrid';
import data from '../../data/list';

const ListItemWidth = 180;

class Demo extends React.Component {
    constructor() {
        super();
        this.state = {
            current: data.current,
            list: data.list,
        };
        this.onSelect = this.onSelect.bind(this);
    }

    onSelect: Function;

    onSelect(id) {
        this.setState({
            current: id,
        });
    }

    render() {
        return (
            <List
                list={this.state.list}
                current={this.state.current}
                onSelect={this.onSelect}
                itemWidth={ListItemWidth}
            />
        );
    }
}

storiesOf('List', module)
    .add('render list', () => (
        <FlexGrid gutter="8px">
            <List
                label="Country"
                labelIcon={<GlobeIcon size={11} />}
                list={data.list}
                current={data.current}
                onSelect={action('onSelect')}
                itemWidth={ListItemWidth}
            />
            <List
                label="Category"
                labelIcon={<FolderIcon size={11} />}
                list={data.list}
                current={data.current}
                onSelect={action('onSelect')}
                itemWidth={ListItemWidth}
            />
        </FlexGrid>
    ))
    .add('render dynamic list', () => <Demo />);
