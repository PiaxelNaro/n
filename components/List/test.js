import React from 'react';
import { shallow, mount } from 'enzyme';

import List, { SubList, ListItem, ListOverlay } from '.';
import data from '../../data/list';

describe('<ListItem>', () => {
    it('render list item', () => {
        const selector = shallow(
            <ListItem isActive={Boolean(true)} isLocked={Boolean(true)} />
        );
        expect(selector).toMatchSnapshot();
    });
});

describe('<SubList>', () => {
    it('render sublist', () => {
        const items = data.list.slice(0, 2);
        const current = items.find(item => item.subList && item.subList.length > 0);
        const onClickSubPanel = jest.fn();
        const onSelectItem = jest.fn();
        const selector = shallow(
            <SubList
                activeSubListId={current.id}
                items={items}
                current={current.id}
                onClickSubPanel={onClickSubPanel}
                onSelect={onSelectItem}
            />
        );
        // current selected item has subList will not be collapsable
        expect(selector).toMatchSnapshot();
        expect(selector.find('.AASubList-icon').prop('isCollapsable')).toEqual(false);

        // trigger button will invoke onClickSubPanel callback
        selector.find('.AASubList-btn').at(0).simulate('click');
        expect(onClickSubPanel.mock.calls[0][0]).toBe(current.id);
        expect(onClickSubPanel.mock.calls[0][1]).toBe(current.subList);

        // trigger ListItem will invoke onSelectItem callback
        selector.find('.AASubList-item').at(0).simulate('click');
        expect(onSelectItem).toHaveBeenCalledWith(items[0].id);
    });

    it('renders collapsable icons', () => {
        expect(shallow(<SubList.Icon isCollapsable />)).toMatchSnapshot();
        expect(shallow(<SubList.Icon />)).toMatchSnapshot();
    });
});

describe('<List>', () => {
    it('render list', () => {
        const current = data.list[0];
        const list = data.list.slice(0, 2);
        const onSelectItem = jest.fn();

        const selector = mount(
            <List list={list} current={current.id} onSelect={onSelectItem} />
        );
        const inst = selector.instance();
        inst.overlay = { closeOverlay: jest.fn() };

        const overlay = shallow(selector.find('OverlayTrigger').prop('getOverlay')());
        expect(overlay).toMatchSnapshot();

        // Close overlay on selection
        overlay.find('SubList').prop('onSelect')(list[0].id);
        expect(onSelectItem).toHaveBeenCalledWith(list[0].id);
        expect(inst.overlay.closeOverlay).toHaveBeenCalled();
    });

    it('render list with label', () => {
        const list = [];
        const selector = shallow(
            <List label="Category" list={list} current={null} onSelect={jest.fn()} />
        );

        expect(selector).toMatchSnapshot();
    });
});

describe('<ListOverlay />', () => {
    it('render list with active subList', () => {
        const current = data.list[1].subList[0];

        const list = data.list.slice(0, 2);
        const onSelectItem = jest.fn();
        const selector = shallow(
            <ListOverlay
                list={list}
                current={current.id}
                onSelect={onSelectItem}
                itemWidth={180}
            />
        );
        selector.instance().componentDidMount();

        expect(selector).toMatchSnapshot();
        expect(selector.find('SubList').length).toEqual(2);
    });

    it('toggle subList', () => {
        const list = data.list.slice(0, 2);
        const current = list.find(item => item.subList && item.subList.length > 0);
        const onSelectItem = jest.fn();
        const selector = shallow(
            <ListOverlay
                list={list}
                current={null}
                onSelect={onSelectItem}
                itemWidth={180}
            />
        );

        selector.find('SubList').prop('onClickSubPanel')(current.id, current.subList);
        expect(selector.find('SubList').at(0).prop('activeSubListId')).toEqual(
            current.id
        );
        expect(selector.find('Overlay').exists()).toBe(true);
    });
});
