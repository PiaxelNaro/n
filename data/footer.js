import languages from './languages';

export default {
    groups: [
        {
            links: [
                {
                    label: '主页',
                    url: '/',
                    newwin: false,
                },
                {
                    label: '销售',
                    url: '/sales',
                    newwin: true,
                }
            ],
        },
        {
            links: [
                {
                    label: '咨询',
                    url: '/consult',
                    newwin: true,
                },
                {
                    label: '关于',
                    url:
                        '/about',
                    newwin: false,
                }
            ],
        },
    ],
    languages_for_switch: languages
};
