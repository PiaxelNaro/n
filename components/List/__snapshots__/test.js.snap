// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List> render list 1`] = `
<Overlay>
  <List__AAList
    gutter="5px"
  >
    <div
      style={
        Object {
          "width": undefined,
        }
      }
    >
      <SubList
        activeSubListId=""
        current="overall"
        items={
          Array [
            Object {
              "id": "overall",
              "title": "Overall",
            },
            Object {
              "id": "games",
              "subList": Array [
                Object {
                  "id": "action",
                  "title": "Action",
                },
                Object {
                  "id": "adventure",
                  "title": "Adventure",
                },
                Object {
                  "id": "arcade",
                  "title": "Arcade",
                },
                Object {
                  "id": "board",
                  "title": "Board",
                },
                Object {
                  "id": "card",
                  "title": "Card",
                },
                Object {
                  "id": "casino",
                  "title": "Casino",
                },
                Object {
                  "id": "dice",
                  "title": "Dice",
                },
                Object {
                  "id": "game_education",
                  "title": "Education",
                },
                Object {
                  "id": "family",
                  "title": "Family",
                },
                Object {
                  "id": "game_kids",
                  "title": "Kids",
                },
                Object {
                  "id": "game_music",
                  "title": "Music",
                },
                Object {
                  "id": "puzzle",
                  "title": "Puzzle",
                },
                Object {
                  "id": "racing",
                  "title": "Racing",
                },
                Object {
                  "id": "role_playing",
                  "title": "Role Playing",
                },
                Object {
                  "id": "simulation",
                  "title": "Simulation",
                },
                Object {
                  "id": "sports",
                  "title": "Sports",
                },
                Object {
                  "id": "strategy",
                  "title": "Strategy",
                },
                Object {
                  "id": "trivia",
                  "title": "Trivia",
                },
                Object {
                  "id": "word",
                  "title": "Word",
                },
              ],
              "title": "Games",
            },
          ]
        }
        onClickSubPanel={[Function]}
        onSelect={[Function]}
      />
    </div>
  </List__AAList>
</Overlay>
`;

exports[`<List> render list with label 1`] = `
<OverlayTrigger
  getOverlay={[Function]}
  trigger="click"
>
  <DropdownButton
    label="Category"
    value=""
  />
</OverlayTrigger>
`;

exports[`<ListItem> render list item 1`] = `
<a
  className="List__ListItem-kf17bn-0 is-active is-locked vJOeA"
/>
`;

exports[`<ListOverlay /> render list with active subList 1`] = `
<Overlay>
  <List__AAList
    gutter="5px"
  >
    <div
      style={
        Object {
          "width": 180,
        }
      }
    >
      <SubList
        activeSubListId=""
        current="action"
        items={
          Array [
            Object {
              "id": "overall",
              "title": "Overall",
            },
            Object {
              "id": "games",
              "subList": Array [
                Object {
                  "id": "action",
                  "title": "Action",
                },
                Object {
                  "id": "adventure",
                  "title": "Adventure",
                },
                Object {
                  "id": "arcade",
                  "title": "Arcade",
                },
                Object {
                  "id": "board",
                  "title": "Board",
                },
                Object {
                  "id": "card",
                  "title": "Card",
                },
                Object {
                  "id": "casino",
                  "title": "Casino",
                },
                Object {
                  "id": "dice",
                  "title": "Dice",
                },
                Object {
                  "id": "game_education",
                  "title": "Education",
                },
                Object {
                  "id": "family",
                  "title": "Family",
                },
                Object {
                  "id": "game_kids",
                  "title": "Kids",
                },
                Object {
                  "id": "game_music",
                  "title": "Music",
                },
                Object {
                  "id": "puzzle",
                  "title": "Puzzle",
                },
                Object {
                  "id": "racing",
                  "title": "Racing",
                },
                Object {
                  "id": "role_playing",
                  "title": "Role Playing",
                },
                Object {
                  "id": "simulation",
                  "title": "Simulation",
                },
                Object {
                  "id": "sports",
                  "title": "Sports",
                },
                Object {
                  "id": "strategy",
                  "title": "Strategy",
                },
                Object {
                  "id": "trivia",
                  "title": "Trivia",
                },
                Object {
                  "id": "word",
                  "title": "Word",
                },
              ],
              "title": "Games",
            },
          ]
        }
        onClickSubPanel={[Function]}
        onSelect={[Function]}
      />
    </div>
  </List__AAList>
</Overlay>
`;

exports[`<SubList> render sublist 1`] = `
<div>
   
  <FlexGrid>
    <FlexGrid.Item
      flex={1}
    >
      <ListItem
        className="AASubList-item"
        isActive={false}
        onClick={[Function]}
      >
        <Text__Text
          ellipsis={true}
        >
          Overall
        </Text__Text>
      </ListItem>
    </FlexGrid.Item>
    <FlexGrid.Item
      flex={0}
    />
  </FlexGrid>
  <FlexGrid>
    <FlexGrid.Item
      flex={1}
    >
      <ListItem
        className="AASubList-item"
        isActive={true}
        onClick={[Function]}
      >
        <Text__Text
          ellipsis={true}
        >
          Games
        </Text__Text>
      </ListItem>
    </FlexGrid.Item>
    <FlexGrid.Item
      flex={0}
    >
      <Button__Blank
        className="AASubList-btn"
        onClick={[Function]}
      >
        <List__Icon
          className="AASubList-icon"
          isCollapsable={false}
        >
          <MdChevronRight
            size={16}
          />
        </List__Icon>
      </Button__Blank>
    </FlexGrid.Item>
  </FlexGrid>
   
</div>
`;

exports[`<SubList> renders collapsable icons 1`] = `
<div
  className="List__Icon-kf17bn-1 dTnsla"
/>
`;

exports[`<SubList> renders collapsable icons 2`] = `
<div
  className="List__Icon-kf17bn-1 lgGwQD"
/>
`;
